write-fonts/src/tables/gpos.rs

```diff
@@ -107,6 +107,27 @@ impl PairPosFormat1 {
             .map(|rec| rec.value_record2.format())
             .unwrap_or(ValueFormat::empty())
     }
+
+    fn check_format_consistency(&self, ctx: &mut ValidationCtx) {
+        let vf1 = self.compute_value_format1();
+        let vf2 = self.compute_value_format2();
+        ctx.in_array(|ctx| {
+            for pairset in &self.pair_sets {
+                ctx.array_item(|ctx| {
+                    ctx.in_field("pair_value_records", |ctx| {
+                        ctx.in_array(|ctx| {
+                            if pairset.pair_value_records.iter().any(|pairset| {
+                                pairset.value_record1.format() != vf1
+                                    || pairset.value_record2.format() != vf2
+                            }) {
+                                ctx.report("all ValueRecords must have same format")
+                            }
+                        })
+                    })
+                })
+            }
+        })
```

Comment:
I don't love the nesting, how might we structure this more like an iter chain where it tends to extend down more than right?

Reply:
Agreed. currently the vast majority of validation code is auto-generated, which makes this less painful. I think that improving the API is out of scope for this PR, but I'll open an issue to look into improving this somewhat. I'm not sure that an iter-like API is possible since it doesn't really support scoping, but we might be able to at least reduce the amount of nesting required.